src/azure-cli-core/azure/cli/core/extension/operations.py

```diff
@@ -306,12 +306,6 @@ def check_version_compatibility(azext_metadata):
 def add_extension(cmd=None, source=None, extension_name=None, index_url=None, yes=None,  # pylint: disable=unused-argument, too-many-statements
                   pip_extra_index_urls=None, pip_proxy=None, system=None,
                   version=None, cli_ctx=None, upgrade=None, allow_preview=None):
-    if allow_preview is None:
```

Comment:
Please add error log to instruct users add `--allow-preview` if an extension has preview version only but users forget adding `--allow-preview` argument.